uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -909,39 +912,37 @@ public static function findOverlappingLeaveRequests($leaveRequestParams, $leaveR
     }
 
     $dateTimeSelector = $isCalculationUnitInHoursOrTOIL
-      ? "lr.from_date < '{$toDate}' AND lr.to_date > '{$fromDate}'"
-      : 'lrd.date BETWEEN %3 AND %4';
+      ? "lr.from_date < %7 AND lr.to_date > %6"
+      : "lrd.date BETWEEN %4 AND %5";
 
-    $absenceTypesIDs = array_merge($absenceTypesIDsWithSameUnit, $absenceTypesIDsWithDifferentUnit);
-    $absenceTypesSelector = $isCalculationUnitInHoursOrTOIL
-      ? (!$isTOIL
-        ? 'AND lr.type_id IN (' . implode(',', $absenceTypesIDsWithSameUnit) . ')'
-        : '')
-      : 'AND lr.type_id IN (' . implode(',', $absenceTypesIDs) . ')';
+    if ($isCalculationUnitInHours) {
+      $absenceTypesSelector = 'AND lr.type_id IN (' . implode(',', $absenceTypesIDsWithSameUnit) . ')';
 
-    $considerRequestsWithDifferentUnit = count($absenceTypesIDsWithDifferentUnit) && !$isTOIL
-      ? "OR ((lrd.date BETWEEN %3 AND %4) AND lr.type_id IN (" .
-        implode(',', $absenceTypesIDsWithDifferentUnit) . "))"
-      : "";
+      if (count($absenceTypesIDsWithDifferentUnit)) {
+        $absenceTypesSelector .=
+          "OR ((lrd.date BETWEEN %4 AND %5) AND lr.type_id IN (" .
+          implode(',', $absenceTypesIDsWithDifferentUnit) . "))";
+      }
+    } else if (!$isTOIL) {
```

Comment:
Not sure I understand this if/else, the first condition is `$isCalculationUnitInHours`, shouldn't it be `$isCalculationUnitInHoursOrTOIL` because the way it is now, if its not toil, it will still pass the first condition.

Reply:
I think there were excessive conditions:

Before: 
```js
if $isCalculationUnitInHoursOrTOIL
  if !$isTOIL "expression"
  else ""
```
is same as 
```
if $isCalculationUnitInHours "expression"
```